src/autodiscovery.js

```diff
@@ -428,13 +429,22 @@ export class AutoDiscovery {
      * be an empty object.
      */
     static async getRawClientConfig(domain) {
+        const default_server_config = SdkConfig.get()['default_server_config'];
+        const base_url_from_config = default_server_config['m.homeserver']['base_url'] || SdkConfig.get()['default_server_name'];
```

Comment:
So we don't change the default element behaviour, using the 'm.homeserver' 'base_url' should be controlled by a config flag e.g. getWellKnownConfigFromHomeServer=true

Reply:
SettingStore config cant be pulled from js-sdk, this change is in js-sdk so this would be only way to change api call by switching domain with server_name